Add KRUSTY 30C Reactivity Insertion Model (#648)
idaholab/virtual_test_bed: doc/content/microreactors/KRUSTY/Griffin-BISON_Multiphysics_30C_Reactivity_Insertion_Test.md

!listing microreactors/KRUSTY/Multiphysics_30C_RIT/KRUSTY_Griffin_SN23_NA23_CMFD_TR.i
         block=Executioner/TimeStepper
         id=input_tr_timestep
         caption=Input block in Griffin main application to control the time step based on the number of iterations and power evolution.

The upper limit for the current time step length between 755 second and 900 s is determined by the postprocessor “power_driven_dt” which is defined in [power_driven_dt].

!listing microreactors/KRUSTY/Multiphysics_30C_RIT/KRUSTY_Griffin_SN23_NA23_CMFD_TR.i
         block=Postprocessors/power_driven_dt
         id=power_driven_dt
         caption=Input block which defines the postprocessor `power_driven_dt`, which is utilized to control the time step size.

These postprocessors help make it possible to control the time step size during the insertion of the second 15 Ȼ.

!listing microreactors/KRUSTY/Multiphysics_30C_RIT/KRUSTY_Griffin_SN23_NA23_CMFD_TR.i
         block=Transfers/from_bison_disp Transfers/to_bison_disp
         id=transfers
         caption=Input blocks to transfer the displacement positions between BISON and Griffin for modeling the second step of the experiment.

> As an essential part of the control approach used for the second stage of reactivity insertion, the shift value of the radial reflector at the last time step must be accessible by the control algorithm. To prevent the confusion caused by fixed point iteration, this value is transferred to the main application and transferred back into another postprocessor.

Reviewer: ```suggestion
As an essential part of the control approach used for the second stage of reactivity insertion, the shift value of the radial reflector at the last time step must be accessible by the control algorithm. To simplify the setup which already includes fixed point iterations between multiple solves, this value is transferred to the main application and transferred back into another `Postprocessor`.
```
please raise an issue in MOOSE if you think there is a confusion there

The input blocks for the three postprocessor, in which the `disp_old_medium` is defined in the Griffin input and the other two defined in BISON input are included in the [bison_pp].

!listing microreactors/KRUSTY/Multiphysics_30C_RIT/KRUSTY_BISON_THERMOMECHANICS_TR.i
         block=Postprocessors/disp_old_origin Postprocessors/disp_old_target
         id=bison_pp

>          caption=Input blocks for defining the postprocessors in transferring the displacements.

Reviewer: ```suggestion
         caption=Input blocks for defining the `Postprocessors` in transferring the displacements.
```

| :- | :- |
| `reflector_disp_init` | initial reflector displacement at the start of the test |
| `reflector_disp_increment_1` | increment of the reflector displacement at the first 15 Ȼ insertion |
| `reflector_disp_increment_2` | nominal total increment of the reflector displacement at the second 15 Ȼ insertion |
| `rdi2_scale_factor` | scale factor for the second 15 Ȼ insertion that is tuned to achieve ~150 second power maintenance |
| `t_first_15c_insertion` | time at which the first 15 Ȼ insertion is completed (starting from t = 0) |
| `t_start_power_monitoring` | time at which the power starts to be monitored for power maintenance during the second 15 Ȼ insertion |
| `existing_reflector_disp` | existing reflector displacement recorded by a post-processor |
| `unit_increment_fraction` | fraction of the nominal total increment of the reflector displacement at the second 15 Ȼ insertion that is applied during each insertion step; the value is obtained by tuning to limit the power jump during each step of the insertion |
| `pw` | total current power of the microreactor |
| `pw_ref` | reference power for power maintenance during the second 15 Ȼ insertion |

> The  function “Ref_mov” has also been used to prescribe the boundary condition for generating the displacement mesh as its input block is included in Listing 6. 

Reviewer: ```suggestion
The `Function` “Ref_mov” has also been used to prescribe the boundary condition for generating the displacement mesh as its input block is included in Listing 6. 
```
